packages/dataviews/src/dataforms-layouts/regular/index.tsx

```diff
@@ -16,9 +16,12 @@ type MemoizedFieldEditProps< Item > = {
 	onChange: Pick< DataFormProps< Item >, 'onChange' >[ 'onChange' ];
 };
 
-const MemoizedFieldEdit = memo( ( { field, value, onChange } ) => (
-	<field.Edit value={ value } field={ field } onChange={ onChange } />
-) ) as < Item >( props: MemoizedFieldEditProps< Item > ) => JSX.Element;
+const MemoizedFieldEdit = memo(
+	( { field, value, onChange } ) => (
+		<field.Edit value={ value } field={ field } onChange={ onChange } />
+	),
+	( prevProps, nextProps ) => prevProps.value === nextProps.value
```

Comment:
I'm not sure we should be doing this though. This would prevent the component from re-rendering if field or onChange change.

Reply:
No for me this PR is good. 

 - I don't expect "field" to change over time
 - I'm not sure what we're doing right now with "onChange" but we can easily pass a stable reference that calls the latest prop.

So I think we can consider memoizing the fields like you did with a simple `memo` call. That said, in general I like to consider metrics before doing performance improvements. If we notice a real issue, we can add a metric to code vitals.run to ensure we keep that good over time.

Reply:
> I don't expect "field" to change over time

One situation where this will happen is when the `elements` of one field depend on another field. Think of countries & regions situation, for example.

Reply:
> One situation where this will happen is when the elements of one field depend on another field. Think of countries & regions situation, for example.

I don't think it's the "field" that changes in this situation, we will need access to the whole "data" though yes and not just the current item.

Reply:
> I don't think it's the "field" that changes in this situation, we will need access to the whole "data" though yes and not just the current item.

🤔 Aren't the filter elements defined by the Field? When a user changes the country to a different one, the region field would need to recreate their element, wouldn't it?

Reply:
> Aren't the filter elements defined by the Field? When a user changes the country to a different one, the region field would need to recreate their element list, wouldn't it?

I think we need a better way to address this use-case. Maybe the elements as a function or something like that. I think field declaration should ideally be "static" mostly.